sos/upload/__init__.py

```diff
@@ -0,0 +1,325 @@
+        width = 80
+        _fmt = ''
+        for line in msg.splitlines():
+            _fmt = _fmt + fill(line, width, replace_whitespace=False) + '\n'
+        return _fmt
+
+    def intro(self):
+        """Print the intro message and prompts for a case ID if one is not
+        provided on the command line
+        """
+        disclaimer = """\
+This utility is used to upload files to a target location \
+based either on a command line option or detecting the local \
+distribution.
+
+The archive to be uploaded may contain data considered sensitive \
+and its content should be reviewed by the originating \
+organization before being passed to any third party.
+
+No configuration changes will be made to the system running \
+this utility.
+"""
+        self.ui_log.info(f"\nsos upload (version {__version__})")
+        intro_msg = self._fmt_msg(disclaimer)
+        self.ui_log.info(intro_msg)
+
+        prompt = "\nPress ENTER to continue, or CTRL-C to quit\n"
+        if not self.opts.batch:
+            try:
+                input(prompt)
+                self.ui_log.info("")
+            except KeyboardInterrupt:
+                self._exit("Exiting on user cancel", 130)
+            except Exception as e:
+                self._exit(e, 1)
+
+    def get_commons(self):
+        return {
+            'cmdlineopts': self.opts,
+            'policy': self.policy,
+            'case_id': self.opts.case_id,
+            'upload_directory': self.opts.upload_directory
+        }
+
+    def set_commons(self, commons):
+        """Set common host data for the Upload targets
+            to reference
+        """
+        self.commons = commons
+
+    def determine_upload_target(self):
+        """This sets the upload target and loads that target's options.
+
+        If no upload target is matched and no target is provided by
+        the user, then we abort.
+
+        If an upload target is provided in the command line,
+        this will not run.
+        """
+        checks = list(self.upload_targets.values())
+        for upload_target in self.upload_targets.values():
+            checks.remove(upload_target)
+            if upload_target.check_distribution():
+                cname = upload_target.__class__.__name__
+                self.ui_log.debug(f"Installation matches {cname}, checking for"
+                                  " upload targets")
+                self.upload_target = upload_target
+                self.upload_name = upload_target.name()
+                self.ui_log.info(
+                    f'Upload target set to {self.upload_name}')
+                break
+
+    def load_upload_targets(self):
+        """Loads all upload targets supported by the local installation
+        """
+        import sos.upload.targets
+        package = sos.upload.targets
+        supported_upload_targets = {}
+        upload_targets = self._load_modules(package, 'targets')
+        for upload_target in upload_targets:
```

Comment:
.. and maybe you can even merge the previous line to this..? `upload_targets is also not used anywhere else.

Reply:
And fixed as well